src/leaflike/core.clj

```diff
@@ -1,24 +1,24 @@
 (ns leaflike.core
-  (require [org.httpkit.server :as httpkit]))
+  (require [leaflike.server :refer [start-server stop-server]]
+           [leaflike.migrations :refer [migrate-db rollback-db]]))
 
 (defn foo
   "I don't do a whole lot."
   [x]
   (println x "Hello, World!"))
 
-(defn app [req]
-  {:status  200
-   :headers {"Content-Type" "application/json"}
-   :body    "Welcome to Leaflike"})
+(defn setup []
+  (migrate-db)
+  (start-server))
 
-(defonce server (atom nil))
+(defn teardown []
+  (rollback-db)
+  (stop-server))
 
-(defn stop-server []
-  (when-not (nil? @server)
-    ;; graceful shutdown: wait 100ms for existing requests to be finished
-    ;; :timeout is optional, when no timeout, stop immediately
-    (@server :timeout 100)
-    (reset! server nil)))
-
-(defn -main [& args]
-  (reset! server (httpkit/run-server #'app {:port 8080})))
+(defn -main
+  [& args]
+  (condp = (first args)
+    "stopapp"  (teardown)
```

Comment:
Calling `(teardown)` from the REPL will stop the server, but `lein run stopapp` will not stop the server. Can you see why?